cumulus/client/consensus/aura/src/collators/slot_based/block_builder_task.rs

```diff
@@ -225,7 +174,7 @@ where
 
 		loop {
 			// We wait here until the next slot arrives.
-			let Ok(para_slot) = slot_timer.wait_until_next_slot().await else {
+			let Some(para_slot) = slot_timer.wait_until_next_slot().await else {
 				return;
 			};
```

Comment:
One thing I realized today, the way we are choosing the the relay parent is not broken. We just take the `best_block_hash` and that is wrong :) We need to take the same relay parent for all the blocks build in the same "relay slot".

So, what I mean is the following:

```
              Relay block X                               x + 1
|----------------------------------------|---------------------------------|
 
Parachain blocks Y - Y + 7
|----|----|----|----|----|----|----|
```

For the parachain blocks Y - Y + 7 we can not change relay parent X. Right now when for example `X'` comes in later and that is better (maybe a primary block), the relay parent would change.

When the slot duration is 12s, switching the relay parent in the middle of the slot is fine (to `X + 1`).

Reply:
Hmm I am not sure that is an issue here. Lets say we see a new best RP at block Y+3, at block opportunity for Y+4 we will check the ancestry and actually produce a fork for the new best block. Which means that during our slot we will produce half our blocks on the new primary block. This sounds desirable to me. 

Why should we commit to building Y - Y + 7 on a RP that is not primary? These blocks can not be included right?

Reply:
We discussed this offline. The issue kind of exists. However, it is not that bad and right now with switching to a better fork is probably a good idea. In the long term we with building on older blocks this will be solved.